### PR TITLE
fix: correctly add and retrieve custom response data from guild

### PR DESCRIPTION
**Pull request by Baconing**
## Summary by CodeRabbit

- **New Features**
	- Enhanced custom response management capabilities.
- **Refactor**
	- Improved internal handling and storage of custom responses.
- **Bug Fixes**
	- Resolved inaccurate response count calculation.

### CODE REVIEW

File: src/main/java/wtf/triplapeeck/oatmeal/commands/essential/NewResponse.java [hunk 1]
```diff
@@ -19,23 +21,19 @@ public void handler(MessageReceivedEvent event, DataCarriage carriage, ThreadMan
...
-            for (CustomResponseData data : list) {
+            for (MariaCustomResponse data : list) {
                 if (Objects.equals(data.getTrigger(), trigger)) {
-                    Main.dataManager.removeCustomResponseData(Long.valueOf(data.getID()));
+                    carriage.guildEntity.removeCustomResponse(data);
                 }
             }
             CustomResponseData data = Main.dataManager.createCustomResponse(trigger, content, carriage.guildEntity);
-            Main.dataManager.saveCustomResponseData(data);
+            carriage.guildEntity.addCustomResponse((MariaCustomResponse) data);
```
[the remark this answers is not in the document]
Comment (Trip-kun) [reply src/main/java/wtf/triplapeeck/oatmeal/commands/essential/NewResponse.java]: Is there not a possibility of a concurrentmodificationexception doing this?

**Review by Trip-kun — APPROVED** [review]
What is difference between ForeignCollection and Collection?

File: src/main/java/wtf/triplapeeck/oatmeal/commands/essential/NewResponse.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Baconing) [reply src/main/java/wtf/triplapeeck/oatmeal/commands/essential/NewResponse.java]: no, there shouldn't be anyway, if there is that would kinda be on your codebase anyway

**Comment by Baconing** [comment]
ForeignCollection is ORMLite specific, and it doesn't work as well anyway. As long as it's properly annotated a normal collection works fine.

File: src/main/java/wtf/triplapeeck/oatmeal/commands/essential/NewResponse.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Trip-kun) [reply src/main/java/wtf/triplapeeck/oatmeal/commands/essential/NewResponse.java]: Okay that's fine then. It need to make sure to save it again afterwards though

[the remark this answers is not in the document]
Comment (Baconing) [reply src/main/java/wtf/triplapeeck/oatmeal/commands/essential/NewResponse.java]: done

**Pull request merged**